Hard to reproduce test failures

In addition to #64, I am also seeing other types of test failures. The key-MS test is a case which often fails with the following pattern

```
 79/114 Test  #83: key-MS ...........................***Failed    0.19 sec
          MOPAC Job: "MS.mop" ended normally on Jun 21, 2022, at 18:40.
WARNING: output file size mismatch, 1032 vs. 1024
Traceback (most recent call last):
  File "/builddir/build/BUILD/mopac-22.0.2/tests/keywords/../run_test.py", line 32, in <module>
    compare_mopac_out_file(out_line, out_list, ref_line, ref_list)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builddir/build/BUILD/mopac-22.0.2/tests/compare_output.py", line 225, in compare_mopac_out_file
    assert ref == out, f'ERROR: string mismatch between {ref} on reference line {ref_line0} and {out} on output line {out_line0}'
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: ERROR: string mismatch between GRADIENT on reference line 388 and NO. on output line 388
```

Unfortunately, I have not been able to access the log files for the failing runs.

--- a/src/output/writmo.F90
+++ b/src/output/writmo.F90
@@ -346,7 +346,7 @@
         end if
       end if
       if (latom == 0) write (iw, '(1X)')
-      if (lprtgra) &
+      if (lprtgra .or. gnorm /= 0.D0) &
         write (iw, '(10X,"GRADIENT NORM           =",F17.5, 10x, "=", 7x, f'//num//'.5, '' PER ATOM'')') &
         gnorm, gnorm/sqrt(1.0*numat)
       if (gnorm > 2.D0 .and. fract > 0.05D0 .and. fract < 1.95D0 .and. &
@@ -1085,7 +1085,7 @@
           vol,  vol*fpc_10*1.d-24
         write(iwrite,*)
       end if
-      if (lprtgra .or. gnorm > 1.D-15 ) write (iwrite, &
+      if (lprtgra .or. gnorm /= 0.D0) write (iwrite, &
         '(  10X, "GRADIENT NORM           =",F17.5, 10x, "=", 7x, f'//num//'.5, " PER ATOM")') gnorm, gnorm/sqrt(1.0*numat)
       if (latom == 0) then
         if (.not.still) then

--- a/src/output/writmo.F90
+++ b/src/output/writmo.F90
@@ -346,7 +346,7 @@
         end if
       end if
       if (latom == 0) write (iw, '(1X)')
-      if (lprtgra .or. gnorm > 1.D-15 ) &
+      if (lprtgra) &
         write (iw, '(10X,"GRADIENT NORM           =",F17.5, 10x, "=", 7x, f'//num//'.5, '' PER ATOM'')') &
         gnorm, gnorm/sqrt(1.0*numat)
       if (gnorm > 2.D0 .and. fract > 0.05D0 .and. fract < 1.95D0 .and. &

